### PR TITLE
Expand documentation on the application functionality

### PR DESCRIPTION
**Pull request by aateem**
This change is [Reviewable](https://reviewable.io/reviews/mirantis/k8s-netchecker-server/40)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +8,24 @@ network connectivity. Connectivity is checked between the server part and multit
 of agents. The agents then make periodical reports to the server and presence of
 reported data indicates aliveness of the link.
 
+How it works
+============
+
+Network checker is an Kubernetes application main purpose of which is checking of connectivity
```
Comment (AlexeyKasatkin) [inline README.md]: a Kubernetes application

File: README.md [hunk 2]
```diff
@@ -6,6 +8,24 @@ network connectivity. Connectivity is checked between the server part and multit
...
+every K8S node using [Daemonset mechanism](https://kubernetes.io/docs/admin/daemons/)
+(to ensure auto-management of the pods). Agents come in two flavors - and there exists two daemonsets
+for each of them. The difference between them is that "Agent-hostnet" is tapped into host network namespace
+via supplying `hostNetwork: True` key-value for corresponding Pod's spec.
+
+The agents then periodically gather network related information
+(e.g. interfaces' info, results of nslookup, etc.) and send formed payload to
+the server address in the K8S cluster's network space.
+
+The server is deployed behind a Kubernetes service.
```
Comment (AlexeyKasatkin) [inline README.md]: What does "behind" mean here?

File: README.md [hunk 3]
```diff
@@ -6,6 +8,24 @@ network connectivity. Connectivity is checked between the server part and multit
...
 
+How it works
+============
+
+Network checker is an Kubernetes application main purpose of which is checking of connectivity
+between the cluster's nodes. Network checker consists of two parts: server (this repository)
+and agent ([developed here](https://github.com/Mirantis/k8s-netchecker-agent)). Agents are deployed on
+every K8S node using [Daemonset mechanism](https://kubernetes.io/docs/admin/daemons/)
+(to ensure auto-management of the pods). Agents come in two flavors - and there exists two daemonsets
+for each of them. The difference between them is that "Agent-hostnet" is tapped into host network namespace
```
Comment (AlexeyKasatkin) [inline README.md]: ... two daemonsets for each of them.

Probably, "... two daemonsets for each agent." is better as it is not clear what is "them" here.

File: README.md [hunk 4]
```diff
@@ -6,6 +8,24 @@ network connectivity. Connectivity is checked between the server part and multit
...
 reported data indicates aliveness of the link.
 
+How it works
+============
+
+Network checker is an Kubernetes application main purpose of which is checking of connectivity
+between the cluster's nodes. Network checker consists of two parts: server (this repository)
+and agent ([developed here](https://github.com/Mirantis/k8s-netchecker-agent)). Agents are deployed on
+every K8S node using [Daemonset mechanism](https://kubernetes.io/docs/admin/daemons/)
+(to ensure auto-management of the pods). Agents come in two flavors - and there exists two daemonsets
```
Comment (AlexeyKasatkin) [inline README.md]: s/exists/exist ?

File: README.md (hunk 1 above)
Comment (aateem) [inline README.md]: Fixed

File: README.md (hunk 4 above)
Comment (aateem) [inline README.md]: Fixed

File: README.md (hunk 3 above)
Comment (aateem) [inline README.md]: Fixed

**Comment by aateem** [comment]
Review status: 0 of 2 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[README.md, line 26 at r1](https://reviewable.io:443/reviews/mirantis/k8s-netchecker-server/40#-KfzxJkkqmYj0yEJc_SA-r1-26:-KgnMFOrOILEwLjNBjgt:b-896fix) ([raw file](https://github.com/mirantis/k8s-netchecker-server/blob/bc769dc68ab0e9c94f1feb1d1b3724f0d294d7b1/README.md#L26)):*
<details><summary><i>Previously, AlexeyKasatkin (Aleksey Kasatkin) wrote…</i></summary><blockquote>

What does "behind" mean here?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mirantis/k8s-netchecker-server/40)*

**Pull request merged**